### ISSUE TITLE
oci_database_pluggable_database_actions does not take the lifecycle state in account when cloning a PDB

### ISSUE THREAD

**Issue by dhoogfr** [oracle/oci-ansible-collection#140]
## Issue Report

When local / remote cloning a PDB, the module checks if the clone PDB already exists as a target.
But it does not look at the lifecycle state of that target.

If a PDB is dropped, it continues to be listed for while with the status "TERMINATED"

```
  - compartment_id: ocid1.compartment.oc1..a xxxxxxxxxxx
    connection_strings: null
    container_database_id: ocid1.database.oc1.xxxxxxxxxxx
    defined_tags: {}
    freeform_tags: {}
    id: ocid1.pluggabledatabase.oc1.xxxxxxxxxx
    is_restricted: null
    lifecycle_details: null
    lifecycle_state: TERMINATED
    open_mode: READ_WRITE
    pdb_name: pdbclone
    time_created: '2021-12-23T08:35:51.174000+00:00'
```

When executing the module to clone with the same target name as the dropped pdb, the module regards the PDB as already existing and does nothing (state ok).
Once (after a couple of hours / day) the the dropped PDB is no longer listed, the module creates the clone.

**Expected behavior**

The module should take the lifecycle_state in consideration when checking if the target PDB already exists or not.

**Environment**

* OS version:

`#45~20.04.1-Ubuntu SMP Wed Nov 10 10:20:10 UTC 2021`

* Ansible version:

```
ansible [core 2.11.6] 
  config file = /home/dhoogfr/exitas/dev/ansible_oci_testing/ansible.cfg
  configured module search path = ['/home/dhoogfr/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/dhoogfr/.local/lib/python3.8/site-packages/ansible
  ansible collection location = /home/dhoogfr/exitas/dev/ansible_oci_testing/collections
  executable location = /home/dhoogfr/.local/bin/ansible
  python version = 3.8.10 (default, Nov 26 2021, 20:14:08) [GCC 9.3.0]
  jinja version = 3.0.3
  libyaml = True
```

* OCI Python SDK version:

`  2.53.1`

* OCI Ansible Modules version:

    insert output of (for oracle.oci):
```
Collection        Version
----------------- -------
ansible.posix     1.3.0  
community.general 4.2.0  
oracle.oci        2.37.0 
```

**Ansible playbook to reproduce the issue**

```
  - name : Clone PDB1
    connection: local
    hosts: localhost
    tasks:

      - name: List pluggable_databases
        oracle.oci.oci_database_pluggable_database_facts:
          database_id: "{{ db_ocid }}"
        check_mode: false

      - name: Local Clone PDB
        oracle.oci.oci_database_pluggable_database_actions:
          action: "local_clone"
          cloned_pdb_name: 'pdbclone'
          pluggable_database_id: "{{ pdb_src_ocid }}"
#          should_pdb_admin_account_be_locked: no
          pdb_admin_password: "{{ pdb_admin_password }}"
          target_tde_wallet_password:  "{{ target_tde_wallet_password }}"
          wait: yes
          wait_timeout: 7200
        register: result

```

**Comment by manojmeda** [comment]
@dhoogfr Thanks for reporting the issue. Looks like a bug to me. We will fix it in the next release.

**Comment by dineshsuthar786** [comment]
@dhoogfr, We have fixed this issue in latest release (v2.38.0). Closing the issue now. Feel free to reopen or create new issue if you still face the problem.
